### PR TITLE
Fix #704, Added stub for CFE_SB_DeletePipe in ut_sb_stubs.c

### LINKED ISSUE #704: ut_sb_stubs.c needs a stub for CFE_SB_DeletePipe
**Is your feature request related to a problem? Please describe.**
CFE_SB_DeletePipe does not have a stub in ut_sb_stubs.c causing undefined reference error when building unit tests for an app that uses this call.

**Describe the solution you'd like**
Add the stub for CFE_SB_DeletePipe with full context tracking.

**Describe alternatives you've considered**
Create my own locally in the app's unit test, but this does not help everyone.

**Requester Info**
Alan Gibson, NASA

### PR DESCRIPTION
**Pull request by asgibson**
**Describe the contribution**
Fixes #704. Added stub for CFE_SB_DeletePipe.
Stub takes PipeId argument and using UT_Stub_CopyFromLocal, saves it into the data buffer.

**Testing performed**
Steps taken to test the contribution:
1. Build steps: 'make ENABLE_UNIT_TESTS=TRUE SIMULATION=native'
Execution steps: 
1. Created simple unit test in cfs_cf calling CFE_SB_DeletePipe directly.
2. PipeId was set as the data buffer pointer and given an initial value.
3. expectedPipeId given a different value than PipeId
4. UtAssert_True used to show they are not equal(fails)
5. Call to CFE_SB_DeletePipe with expectedPipeId as argument
6. UtAssert_True used to show they are equal(pass)
7. Ran unit test from build/native/apps/cfs_cf/unit-test
Please note: this test was not saved to any repo.
Output:
[BEGIN] 04 Test_check_sb_stubs
[ FAIL] 04.001 cf_app_tests.c:77 - PipeId = 0x00000005, exp = 0x00000006
[ PASS] 04.002 cf_app_tests.c:83 - PipeId = 0x00000006, exp = 0x00000006
[  END] 04 Test_check_sb_stubs  TOTAL::2     PASS::1     FAIL::1      MIR::0      TSF::0      N/A::0  

**Expected behavior changes**
A clear and concise description of how this contribution will change behavior and level of impact.
 - API Change: cFE ut_sb_stubs now has CFE_SB_DeletePipe available.
 - Behavior Change: App unit tests requiring this will not fail to build due to undefined reference to `CFE_SB_DeletePipe'

**System(s) tested on**
 - Hardware: PC
 - OS: RHEL-7.6 (Maipo) Linux 3.10.0-1062.1.2.el7.x86_64
 - Versions: cFE 6.7

**Additional context**
None.

**Third party code**
None.

**Contributor Info - All information REQUIRED for consideration of pull request**
Alan Gibson, NASA, GSFC-0587

### CODE REVIEW

**Comment by asgibson** [comment]
My apologies, I forgot to test that CFE_SUCCESS was returned by the stub so I checked that too.
[BEGIN] 04 Test_check_sb_stubs
[ FAIL] 04.001 cf_app_tests.c:78 - PipeId = 0x00000005, exp = 0x00000006
[ PASS] 04.002 cf_app_tests.c:84 - CFE_SB_DeletePipe returned 0x00000000, and should have been 0x00000000 (CFE_SUCCESS)
[ PASS] 04.003 cf_app_tests.c:85 - PipeId = 0x00000006, exp = 0x00000006
[  END] 04 Test_check_sb_stubs  TOTAL::3     PASS::2     FAIL::1      MIR::0      TSF::0      N/A::0   

[  END] 04 SUMMARY              TOTAL::6     PASS::5     FAIL::1      MIR::0      TSF::0      N/A::0

File: fsw/cfe-core/ut-stubs/ut_sb_stubs.c [hunk 1]
```diff
@@ -158,6 +158,35 @@ int32 CFE_SB_CreatePipe(CFE_SB_PipeId_t *PipeIdPtr, uint16 Depth,
...
+**
+** \par Description
+**        This function is used to mimic the response of the cFE SB function
+**        CFE_SB_DeletePipe.
+**
+** \par Assumptions, External Events, and Notes:
+**        None
+**
+** \returns
+**        Returns CFE_SUCCESS.
```
Comment (skliper) [inline fsw/cfe-core/ut-stubs/ut_sb_stubs.c:173]: Unless configured not to?

Comment (asgibson) [reply fsw/cfe-core/ut-stubs/ut_sb_stubs.c:173]: I agree with that.
Is that what this means?
```
** \returns
**        Returns either a user-defined status flag or CFE_SUCCESS.
```
Which is the only example I could see that would be comparable.

Comment (skliper) [reply fsw/cfe-core/ut-stubs/ut_sb_stubs.c:173]: Yes, that's a better description.

**Comment by asgibson** [comment]
Changed comment for return type on CFE_SB_DeletePipe.

**Comment by astrogeco** [comment]
CCB 2020-05-27 APPROVED

**Pull request merged**